src/test/java/com/fakng/fakngagrgtr/parser/UberParserTest.java

```diff
@@ -0,0 +1,86 @@
+package com.fakng.fakngagrgtr.parser;
+
+import com.fakng.fakngagrgtr.parser.uber.UberParser;
+import com.fakng.fakngagrgtr.persistent.company.Company;
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+public class UberParserTest extends AbstractParserTest {
+
+    private static final String URL = "Uber";
+
+    private UberParser uberParser;
+
+    @Mock
+    private CompanyRepository companyRepository;
+
+    @Mock
+    private LocationProcessor locationProcessor;
+
+    @BeforeEach
+    public void init() throws IOException {
+        String json = getTestData("uber-data.json");
+        uberParser = new UberParser(mockWebClient(json), companyRepository, locationProcessor, URL);
+    }
+
+    @Test
+    public void testParseCreatesVacanciesFromJsonWithCachedLocations() {
+        Company company = prepareCompany();
+        Mockito.when(companyRepository.findByTitle(URL))
+                .thenReturn(Optional.of(company));
+        Mockito.when(locationProcessor.processLocation(company, "city_1", "country_1"))
+                .thenReturn(company.getLocations().get(0));
+        Mockito.when(locationProcessor.processLocation(company, "city_2", "country_2"))
+                .thenReturn(company.getLocations().get(1));
+        uberParser.init();
+        List<Vacancy> expectedVacancies = prepareVacancies(company);
+
+        List<Vacancy> actualVacancies = uberParser.parse();
+
+        assertEquals(expectedVacancies.size(), actualVacancies.size());
+        for (int i = 0; i < expectedVacancies.size(); i++) {
+            assertVacancy(expectedVacancies.get(i), actualVacancies.get(i));
+        }
+    }
+
+    private List<Vacancy> prepareVacancies(Company company) {
```

Comment:
Maybe move the definition of this method to AbstractPartserTest without implementation? So that no one forgets to define it, and so that everyone has the same. And then each of the tests will determine it in its own way.

Reply:
Different parsers can have their own format and different result, so I think there is no need to move it to the base class.